projects/plugins/jetpack/extensions/shared/components/gutenberg/context/resolver.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { select } from '@wordpress/data';
+
+export const SITE_EDITOR_CONTEXT = 'SITE_EDITOR_CONTEXT';
+export const WIDGET_EDITOR_CONTEXT = 'WIDGET_EDITOR_CONTEXT';
+export const POST_EDITOR_CONTEXT = 'POST_EDITOR_CONTEXT';
+export const CUSTOMIZER_EDITOR_CONTEXT = 'CUSTOMIZER_EDITOR_CONTEXT';
+export const NAVIGATION_EDITOR_CONTEXT = 'NAVIGATION_EDITOR_CONTEXT';
+export const UNKNOWN_EDITOR_CONTEXT = 'UNKNOWN_EDITOR_CONTEXT';
+
+export const getGutenbergContext = () => {
```

Comment:
This should be moved to `shared/get-gutenberg-context.js`.

As an aside, it might be perceived as a hook, but it doesn't really use the selectors, it only checks for their existence, so I'm fine with `get` rather than `use`. 🙂

Reply:
Actually, what do you think to keep it super simple and rename it as `getEditorType` (and `shared/get-editor-type.js`, and all the `_CONTEXT`)?

We could also go as far as replacing the constants with simpler hardcoded strings like `post`, `site`, `widget`, `customizer`, `navigation`, and undefined.
All in all, we are already directly coupling this with hardcoded store names, and the editor type names are very recognizable in the Gutenberg world (maybe with the exception of "customizer", which is a weird Gutenberg case, but the name would be well understood in WP).

I mean something like this:

```js
export function getEditorType() {
  if ( !! select( 'core/edit-site' ) {
    return 'site';
  }

  // ...

  return undefined;
}
```

Reply:
Updated the name and location of the file. Also removed the _CONTEXT from the constants. I'm not going to use strings since then we would get many more results when trying to find usages of these editor types.